### ISSUE TITLE
No iteration was executed, please check the data reader

### ISSUE THREAD

**Issue by liu824** [PaddlePaddle/models#4731]
训练时我遇到下面的问题：
[INFO: train.py:  249]: Namespace(batch_size=None, config='models-1.6/PaddleCV/PaddleVideo/configs/bmn.yaml', epoch=None, fix_random_seed=False, learning_rate=None, log_interval=10, model_name='BMN', no_memory_optimize=False, pretrain=None, resume=None, save_dir='./data/checkpoints', use_gpu=True, valid_interval=1)
[INFO: config_utils.py:   70]: ---------------- Train Arguments ----------------
[INFO: config_utils.py:   72]: MODEL:
[INFO: config_utils.py:   74]:     name:BMN
[INFO: config_utils.py:   74]:     tscale:551
[INFO: config_utils.py:   74]:     dscale:551
[INFO: config_utils.py:   74]:     feat_dim:1024
[INFO: config_utils.py:   74]:     prop_boundary_ratio:0.5
[INFO: config_utils.py:   74]:     num_sample:2
[INFO: config_utils.py:   74]:     num_sample_perbin:3
[INFO: config_utils.py:   74]:     anno_file:trainnew.json
[INFO: config_utils.py:   74]:     feat_path:work/trainnew/videofeature
[INFO: config_utils.py:   72]: TRAIN:
[INFO: config_utils.py:   74]:     subset:train
[INFO: config_utils.py:   74]:     epoch:9
[INFO: config_utils.py:   74]:     batch_size:2
[INFO: config_utils.py:   74]:     num_threads:8
[INFO: config_utils.py:   74]:     use_gpu:True
[INFO: config_utils.py:   74]:     num_gpus:4
[INFO: config_utils.py:   74]:     learning_rate:0.001
[INFO: config_utils.py:   74]:     learning_rate_decay:0.1
[INFO: config_utils.py:   74]:     lr_decay_iter:4200
[INFO: config_utils.py:   74]:     l2_weight_decay:0.0001
[INFO: config_utils.py:   72]: VALID:
[INFO: config_utils.py:   74]:     subset:validation
[INFO: config_utils.py:   74]:     batch_size:16
[INFO: config_utils.py:   74]:     num_threads:8
[INFO: config_utils.py:   74]:     use_gpu:True
[INFO: config_utils.py:   74]:     num_gpus:4
[INFO: config_utils.py:   72]: TEST:
[INFO: config_utils.py:   74]:     subset:validation
[INFO: config_utils.py:   74]:     batch_size:1
[INFO: config_utils.py:   74]:     num_threads:1
[INFO: config_utils.py:   74]:     snms_alpha:0.001
[INFO: config_utils.py:   74]:     snms_t1:0.5
[INFO: config_utils.py:   74]:     snms_t2:0.9
[INFO: config_utils.py:   74]:     output_path:data/output/EVAL/BMN_results
[INFO: config_utils.py:   74]:     result_path:data/evaluate_results
[INFO: config_utils.py:   72]: INFER:
[INFO: config_utils.py:   74]:     subset:test
[INFO: config_utils.py:   74]:     batch_size:1
[INFO: config_utils.py:   74]:     num_threads:1
[INFO: config_utils.py:   74]:     snms_alpha:0.4
[INFO: config_utils.py:   74]:     snms_t1:0.5
[INFO: config_utils.py:   74]:     snms_t2:0.9
[INFO: config_utils.py:   74]:     filelist:data/dataset/bmn/infer.list
[INFO: config_utils.py:   74]:     output_path:data/output/INFER/BMN_results
[INFO: config_utils.py:   74]:     result_path:data/predict_results
[INFO: config_utils.py:   75]: -------------------------------------------------
W0701 10:13:39.680063   368 device_context.cc:252] Please NOTE: device: 0, CUDA Capability: 70, Driver API Version: 9.2, Runtime API Version: 9.0
W0701 10:13:39.683766   368 device_context.cc:260] device: 0, cuDNN Version: 7.3.
train subset video numbers: 3500
validation subset video numbers: 0
[INFO: bmn_proposal_metrics.py:   62]: Resetting train metrics...
[INFO: bmn_proposal_metrics.py:   62]: Resetting valid metrics...
[INFO: train_utils.py:   45]: ------- learning rate [0.], learning rate counter [-1] -----
[INFO: train_utils.py:  104]: No iteration was executed, please check the data reader
我的配置文件如下：
MODEL:
    name: "BMN"
    tscale: 551
    dscale: 551
    feat_dim: 1024
    prop_boundary_ratio: 0.5
    num_sample: 2
    num_sample_perbin: 3
    anno_file: "trainnew.json"
    feat_path: 'work/trainnew/videofeature'

TRAIN:
    subset: "train"
    epoch: 9
    batch_size: 2
    num_threads: 8
    use_gpu: True
    num_gpus: 4
    learning_rate: 0.001
    learning_rate_decay: 0.1
    lr_decay_iter: 4200
    l2_weight_decay: 1e-4
其他没有变化

**Comment by huangjun12** [comment]
检查下数据文件路径呢

**Comment by liu824** [comment]
路径应该没问题，你看它都读取到了我的训练集为3500.验证集为0

**Comment by liu824** [comment]
我猜测是不是这个必须有验证集才能跑啊，虽然我已经把valid_interval设置为0了

**Comment by huangjun12** [comment]
> 我猜测是不是这个必须有验证集才能跑啊，虽然我已经把valid_interval设置为0了

不是的，
看报错是reader没有返回数据；
train subset video numbers: 3500
这两句只是在实例化reader时，从anno_file里读取到的信息

**Comment by liu824** [comment]
是因为找不到我的数据集吗？

**Comment by huangjun12** [comment]
https://github.com/PaddlePaddle/models/blob/release/1.8/PaddleCV/video/reader/bmn_reader.py
可能是 你在这个reader里定位下吧

**Comment by liu824** [comment]
定位是指？单步调试？

**Comment by huangjun12** [comment]
先检查下feat_path是否存在；feat_path与anno_file里的文件名是否对应；
确认没问题在reader里调试看看

**Comment by liu824** [comment]
好的

**Comment by liu824** [comment]
![image](https://user-images.githubusercontent.com/55372277/86224984-2ce9ff80-bbbc-11ea-9bb7-88d4cfb6781b.png)
数据都是对的

**Comment by liu824** [comment]
我的json文件相比于例程里少了一些参数，虽然我看源码并没有用到这些参数，但不确定是否是因为这个原因导致错误
![image](https://user-images.githubusercontent.com/55372277/86225505-e8ab2f00-bbbc-11ea-9724-1b208171f1c1.png)

**Comment by huangjun12** [comment]
没有具体代码和使用数据难以定位，只能告知可能原因，不知你是否已定位出问题

**Comment by liu824** [comment]
您好，这个问题您刚才已经告诉我答案了，是因为batch_size的数量小于GPU的数量造成的
